### ISSUE TITLE
What proteins are in the ERAD BiologicalProcess/Pathway? - Two Actors Respond

### ISSUE THREAD

**Issue by sstemann** [NCATSTranslator/testing#38]
**Query:** [erad] (https://github.com/NCATSTranslator/testing/blob/main/ars-requests/not-none/erad.json)
**PK:** 8c401fb4-922f-4c49-adfe-436314838c74
GO:0036503
[**Results Tracking Sheet**] ((https://docs.google.com/spreadsheets/d/1Y2yfUFl2SIgv5KkyxdQB94UE3HtVdJNJ7KdnwQgZcYI/edit?usp=sharing)

![image](https://user-images.githubusercontent.com/38321826/113624130-3dd50f00-962d-11eb-9708-1c7eb715aeac.png)

**Responses From:**
- ARAX (Expander Agent)
- Unsecret Agent

Note:  We also tried it with flipped Object/Subject (ARAGORN returned 8 copies of "protein".  We also tried it with Biolink category:pathway and got the same results set.

**Comment by cbizon** [comment]
ranking-agent lumps genes and proteins together under Gene.  So if n1.category = "biolink:Gene" then we return results.  Also if n1.category = "biolink:GeneOrGeneProduct".  As long as ARAs can handle biolink subclassing, using that union might lead to results from more of them, since it should work whether ARAs use genes or proteins..

**Comment by sstemann** [comment]
> ranking-agent lumps genes and proteins together under Gene. So if n1.category = "biolink:Gene" then we return results. Also if n1.category = "biolink:GeneOrGeneProduct". As long as ARAs can handle biolink subclassing, using that union might lead to results from more of them, since it should work whether ARAs use genes or proteins..

@cbizon  still two responses:  a0fe5554-4c2e-472b-9cf6-6f0cc5556de2
but this time Aragorn and CAM, so not sure if the goal of subclassing is achieved? i tried it with a node array, and it got the original two and the Aragorn and CAM - 15bf580c-a53b-477e-9d2b-bb25c35c2cd6 - so I'm not sure if this is an ARS requirement to generalize with arrays or if this should be handled somewhere else?

**Comment by brettasmi** [comment]
5c3d4b40-8da2-42db-ab9c-bc572d5d63e4:
```
{
  "message": {
    "query_graph": {
      "edges": {
        "e01": {
          "object": "n0",
          "subject": "n1"
        }
      },
      "nodes": {
        "n0": {
          "category": "biolink:BiologicalProcess",
          "id": "GO:0036503"
        },
        "n1": {
          "category": "biolink:Gene"
        }
      }
    }
  }
}
```
and 
c0ec4c9a-f109-4ca0-87ba-22ca195dfcfc:
```
{
  "message": {
    "query_graph": {
      "edges": {
        "e01": {
          "object": "n0",
          "subject": "n1"
        },
        "e02": {
          "object": "n1",
          "subject": "n2"
        }
      },
      "nodes": {
        "n0": {
          "category": "biolink:BiologicalProcess",
          "id": "GO:0036503"
        },
        "n1": {
          "category": "biolink:Gene"
        },
        "n2": {
          "category": "biolink:Protein"
        }
      }
    }
  }
}
```
Should work for us, but for whatever reason, I'm having trouble seeing the results from the ARS right now.

**Comment by southalln** [comment]
I'd like to come back to the point "ranking-agent lumps genes and proteins together under Gene". In the original FOA we had written something about expectations for the ARAs to the effect "the autonomous relay agents should adroitly handle the integration of knowledge from multiple Knowledge Providers and thus multiple different domains of biomedical knowledge." I think it is reasonable for the user to pose the question requesting proteins, and then the expectation would be for the ARA to know that KPs including whatever resources Ranking agent calls out to require a gene identifier in lieu of a protein identifier.

**Comment by sstemann** [comment]
@brettasmi @cbizon if the TRAPI query is written such that:

"n1": {
              "category": [
                "biolink:GeneOrGeneProduct",
                "biolink:Protein",
                "biolink:Gene"
              ]
            }

then yes, many ARAs return responses (8b63edcf-3c04-480a-889c-6d04029354ac)

**Comment by cbizon** [comment]
@sstemann that's interesting.  You shouldn't have to include gene and protein as well.  I would consider not responding to the superclass biolink:GeneOrGeneProduct a bug in the ARA.

@southalin: yeah.  Maybe another way to say what you are saying is: if ranking agent is going to mush gene and protein information together, then it should also respond to queries in the same way (consider gene and protein queries as queries for the same thing).  I am hoping that the discussion of formal conflation that Mike Bada instigated and is simmering in the data modeling meetings will provide a more complete solution to this issue.

**Comment by sstemann** [comment]
@cbizon i like the sounds of that but the results appear to be different and without biolink:Gene (93aa46eb-6613-465e-85ff-4e635ccbb6d8) we lost results from BTE and Improve:

![image](https://user-images.githubusercontent.com/38321826/113728592-39a80080-96c4-11eb-8b87-043d77b4cbc8.png)

**Comment by MarkDWilliams** [comment]
Yeah, I'm not sure how consistently the biolink superclasses (GeneOrGeneProduct, DiseaseOrPhenotypicFeature) are implemented across the ARAs.  Sounds like something that's worth touching on briefly on today's standup.

**Comment by cbizon** [comment]
Yep.   Tests for this are part of the /testing/onehops framework for what it's worth.

**Comment by balhoff** [comment]
@Shalsh23 asked me about this query for CAM-KP. We have two issues with it working correctly: (1) apparently we need GeneOrGeneProduct instead of Protein, and (2) we have a performance issue with unbound predicates; it works with `related_to` (this is our bug).

This query works with CAM-KP:

```json
{
  "message": {
    "query_graph": {
      "edges": {
        "e01": {
          "subject": "n1",
          "object": "n0",
"predicate": "biolink:related_to"
        }
      },
      "nodes": {
        "n0": {
          "id": "GO:0036503",
          "category":"biolink:BiologicalProcess"
                },
        "n1": {
          "category": "biolink:GeneOrGeneProduct"
        }
      }
    }
  }
 }
```

**Comment by colleenXu** [comment]
I ran the updated query below through the ARS, PK: 222d4dd6-a246-473e-9fbd-be9f06e7e154

![Screen Shot 2021-09-22 at 1 16 38 PM](https://user-images.githubusercontent.com/43731687/134415596-5659463b-01b7-4217-bf06-153a788ff2b3.png)


The updated query could be something like this (it's missing the workflow section but the rest is TRAPI v1.2):
```
{
    "message": {
        "query_graph": {
            "nodes": {
                "n0": {
                    "ids": ["GO:0036503"],
		            "categories":["biolink:BiologicalProcess"]
                },
                "n1": {
                    "categories": ["biolink:Gene", "biolink:Protein"]
                }
            },
            "edges": {
                "e0": {
                    "subject": "n0",
                    "object": "n1"
                }
            }
        }
    }
}
```
